### ISSUE TITLE
Too much time to compile files

### ISSUE THREAD

**Issue by MACruzDeveloper** [Kuniwak/gulp-dependencies-changed#10]
Hi,

I am testing this plugin in a folders structure with jade files (with includes, mixins... and dependencies between them):

-jade
     --mixins
     --includes
     --layouts
     --views

It seems to work perfectly when the match only affect a view, but as I add more matches to more views, when I change some file in a subfolder, the time it takes to compile the view increases coming to take 9-10 seconds to change.. Is this behavior normal?? Is there any solution to decrease the time?

```
gulp.task('compileJade', function(){
    return gulp.src('app/jade/views/**/*.jade')
        .pipe(changed({
          matcher: /#include\s+"(.+)"/g,
          dest: function(srcRelPath, srcBasePath) {
            return rename(srcRelPath, function(fileObj) {
              return {
                dirname: Path.join('app/html/' + fileObj.dirname),
                extname: '.html'
              };
            });
          },
        }))
        .pipe(jade()
        .pipe(gulp.dest('app/html'))
});
```

`gulp.watch('app/jade/**/*.jade', ['compileJade']);`

Thanks!

**Comment by Kuniwak** [comment]
@TxikiCruz 

Thanks for reporting.

Could you try the [`debug` option](https://github.com/mixi-inc/gulp-dependencies-changed#debug) please?
This plugin can print debug output if it is `true`.
I want to see the debug output.

**Comment by TxikiCruz** [comment]
@Kuniwak

Debug print this code for every file: 

Related files ["/app/jade/views/values/stocks-see-favourites.jade"] (from "/app/jade/views/values/stocks-see-favourites.jade")

But in a file that have changed debug print: 

File "/app/jade/views/cards/hire-a-new-card.jade" is newer than "/app/html/cards/hire-a-new-card.html"

I have a general file where I put as dependencies all the files I have in my folders jade:

#include "../includes/cards/hire-new-card-step-01.jade"
#include "../mixins/list/index.jade"
...

Could the problem be with this file??

Thanks.

**Comment by Kuniwak** [comment]
I think that some circular dependencies are included in your code.

Because your debug output says that "`stocks-see-favourites.jade` is depended by itself".
(NOTE: `Related files "XXX" (from "YYY")` means that "XXX depended by YYY")

Could you resolve the circular dependencies?

**Comment by MACruzDeveloper** [comment]
You are right, some of dependencies are circular as you say, and I can see another problem now, the dependencies multiply every time I add a match in a view. 

I think my tasks are not right, I will try to find the right way, thanks!
